### ISSUE TITLE
the avoid_redundant_argument_values lint doesn't always report consistently

### ISSUE THREAD

**Issue by devoncarew** [dart-lang/linter#2073]
**Describe the issue**
A clear and concise description of what the issue is.

I'm seeing issues with when `avoid_redundant_argument_values` fires. Sometime it won't fire when it should, but doing something like leaving the file and re-opening it will cause the lint to then report.

In one case, I was not seeing it report locally, but was seeing it report on a CI (https://travis-ci.org/github/flutter/devtools/jobs/677064517#L497). Later edits to the file / file navigation then caused it to report locally as well.

**Comment by bwilkerson** [comment]
I don't see anything in the implementation of the lint that seems like a likely cause of this behavior. I have to suspect, then, that the problem is really on the analyzer/server side. Two possible causes:
- it's possible that the constant evaluation is working differently depending on cached values, and
- it's possible that sometimes server is pulling diagnostics from the cache when it needs to re-analyze the file.

@scheglov For additional thoughts

**Comment by pq** [comment]
👍 to what @bwilkerson said.

For reference, we had a similar issue reported (and possibly fixed?) in https://github.com/dart-lang/linter/issues/1744.

**Comment by devoncarew** [comment]
I can report again if I see this, in the chance that this was a fluke (or user error).

**Comment by pq** [comment]
I doubt it was user error (or what that would even mean) -- analysis should _just work_ and **_consistently_**.  Definitely bump this if you hit this again and have any insights about a repro.

**Comment by scheglov** [comment]
The lint uses `constantValue` that is not computed if analyzer does not have to.

It is very easy to reproduce not reporting the hint:
`test.dart`
```dart
import 'test2.dart';

main() {
  foo(a: 42);
}
```


`test2.dart`
```dart
void foo({int a = 42}) {}
```

I'm going to deprecate and remove it.
https://dart-review.googlesource.com/c/sdk/+/144666

**Comment by pq** [comment]
OMG.  Amazing.  Thanks @scheglov!

Update to use `computeConstantValue()` here: https://github.com/dart-lang/linter/pull/2077
